### ISSUE TITLE
Add default docker environment

### ISSUE THREAD

**Issue by hickeng** [vmware/vic#725]
Docker adds default environment to every container. This needs to be merged with the config specified on the command line and in the layer config before calling create on the port layer. This interacts with #411.

The defaults [are detailed here](https://docs.docker.com/engine/reference/run/#env-environment-variables):
```
#Set based on the value of USER
HOME=... 

#The truncated id of the container
HOSTNAME=... 

PATH="/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"

# if the container is allocated a pseudo-TTY
TERM="xterm" 
```

**Comment by mdubya66** [comment]
Is this what is causing our integration tests to fail for the last 3 days?

**Comment by mdubya66** [comment]
This might be the original issue zach. Close whichever one you feel like.

**Comment by jzt** [comment]
@hickeng It looks like `TERM=xterm` and `PATH` get injected cleanly. I've added a line to inject the truncated container id as `HOSTNAME` into the metadata once the container is created. I have two questions:

First, I'm still not sure about `HOME` yet, though - what are our options with regard to the bootstrap iso? We could mimic docker behavior in the following ways:
* if we provide a `-u <UID>` option, we could default `HOME` to `/`
* if we provide a username that doesn't exist in the container to `-u`, we could throw an error: `docker: Error response from daemon: linux spec user: unable to find user zach: no matching entries in passwd file.`
if we provide a user that exists in the container, we can derive and set `HOME` from `/etc/passwd`:
  
  ```
$ docker run -it -u nobody busybox
/ $ id
uid=99(nobody) gid=99(nogroup)
/ $ echo $HOME
/home
/ $ cat /etc/passwd
root:x:0:0:root:/root:/bin/sh
daemon:x:1:1:daemon:/usr/sbin:/bin/false
bin:x:2:2:bin:/bin:/bin/false
sys:x:3:3:sys:/dev:/bin/false
sync:x:4:100:sync:/bin:/bin/sync
mail:x:8:8:mail:/var/spool/mail:/bin/false
www-data:x:33:33:www-data:/var/www:/bin/false
operator:x:37:37:Operator:/var:/bin/false
nobody:x:99:99:nobody:/home:/bin/false
```

Second, how should we handle the part of the `PATH` that does not exist on the container's filesystem? We mentioned creating directories that need to exist in the scratch layer when it (the scratch layer) gets created , but I'm not totally clear on where we landed with regard to that. There's also the whole issue of checking `Os` in the container config and setting the default environment variables based on that.
